src/js/components/MultiSelect/MultiSelectContainer.js

```diff
@@ -0,0 +1,640 @@
+          activeIndex >= 0 &&
+          activeIndex < options.length
+        ) {
+          event.preventDefault(); // prevent submitting forms
+          selectOption(activeIndex)(event);
+        }
+      },
+      [activeIndex, selectOption, options, isDisabled],
+    );
+
+    const customSearchInput = theme.select.searchInput;
+    const SelectTextInput = customSearchInput || TextInput;
+    const selectOptionsStyle = theme.select.options
+      ? {
+          ...theme.select.options.box,
+          ...theme.select.options.container,
+        }
+      : {};
+
+    // handle when limit is reached
+    useEffect(() => {
+      if (value.length === limit) {
+        const newDisabled = [...disabledProp];
+        // disable everything that is not selected
+        const originallyDisabled = (index) => {
+          const option = allOptions[index];
+          let result;
+          if (disabledKey) {
+            result = applyKey(option, disabledKey);
+          } else if (Array.isArray(disabledProp)) {
+            if (typeof disabledProp[0] === 'number') {
+              result = disabledProp.indexOf(index) !== -1;
+            } else {
+              result = disabledProp.indexOf(optionValue(index)) !== -1;
+            }
+          }
+          return result;
+        };
+        for (let i = 0; i < options.length; i += 1) {
+          if (!isSelected(i) && !originallyDisabled(i)) {
+            newDisabled.push(options[i]);
+          }
+        }
+        if (usingKeyboard)
+          setShowA11yLimit('Selected. Maximum selection limit reached.');
+        setDisabled(newDisabled);
+      } else {
+        if (usingKeyboard) setShowA11yLimit(undefined);
+        setDisabled(disabledProp);
+      }
+    }, [
+      isSelected,
+      optionValue,
+      value,
+      limit,
+      disabledProp,
+      allOptions,
+      disabledKey,
+      labelKey,
+      options,
+      usingKeyboard,
+      valueKey,
+    ]);
+
+    // reset showA11yLimit after announcement is read
+    useEffect(() => {
+      if (showA11yLimit !== undefined) {
+        setTimeout(() => {
+          setShowA11yLimit(undefined);
+        }, 2000);
+      }
+    }, [showA11yLimit]);
+
+    const selectedValuesDisabled = useCallback(() => {
+      let disabledSelected = 0;
+      for (let i = 0; i < allOptions.length; i += 1) {
+        if (value.includes(optionValue(i)) && isDisabled(i))
+          disabledSelected += 1;
+      }
+      if (value.length === disabledSelected) return true;
+      return false;
+    }, [value, allOptions, isDisabled, optionValue]);
+
+    const selectClearAll =
+      search === '' || search === undefined ? (
+        <>
+          <Box alignSelf="center">
+            {value.length === 0 ? (
+              <Text size="small">0 selected</Text>
+            ) : (
+              <Text size="small">
+                {value.length} selected of {options.length}
+              </Text>
+            )}
+          </Box>
+          <Box>
```

Comment:
why need this `<Box>` can we just use a `<>`?

Reply:
Fixed